mining/mining.go

```diff
@@ -620,14 +627,38 @@ mempoolLoop:
 		switch {
 		// If segregated witness has not been activated yet, then we
 		// shouldn't include any witness transactions in the block.
-		case tx.MsgTx().HasWitness() && !includeWitness:
+		case tx.MsgTx().HasWitness() && !segwitActive:
 			continue
 
 		// Otherwise, Keep track of if we've included a transaction
 		// with witness data or not. If so, then we'll need to include
 		// the witness commitment as the last output in the coinbase
 		// transaction.
-		case tx.MsgTx().HasWitness() && includeWitness:
+		case tx.MsgTx().HasWitness() && segwitActive:
+			// If we're about to include a transaction bearing
+			// witness data, then we'll also need to include a
+			// witness commitment in the coinbase transaction.
+			// Therefore, we account for the additional weight
+			// within the block.
+			if !witnessIncluded {
+				// First we account for the additional witness
+				// data in the witness nonce of the coinbaes
+				// transaction: 32-bytes of zeroes.
+				blockWeight += 2 + 32
```

Comment:
It just occurred to me that I was actually double counting the weight of the coinbase transaction with this fragment. This is due to the fact that the weight of the coinbase transaction without a witness commitment was already computed above. So what I need to do instead is add the _difference_ of the two coinbase transactions (with commitment vs without). 

Modified the code to craft a model coinbase transaction within a witness commitment, then the diff of that vs the current coinbase transaction is added to the total block weight.